dotnet/StachExtensions/FactSet.Protobuf.Stach.Extensions/V2/RowOrganizedStachExtension.cs

```diff
@@ -136,8 +141,25 @@ private Models.Table GenerateTable(RowOrganizedPackage.Types.Table table)
 
             foreach (var metadataItem in table.Data.TableMetadata)
             {
-                var metadataValue = StachUtilities.ValueToString(metadataItem.Value.Value);
-                finalTable.Metadata.Add(metadataItem.Key, metadataValue);
+                var metadataValueString = StachUtilities.ValueToString(metadataItem.Value.Value);
+                var metadataValue = metadataItem.Value.Value;
+
+                finalTable.Metadata.Add(metadataItem.Key, metadataValueString);
+
+                List<Value> valuesList = new List<Value>();
+                if (metadataValue.KindCase == Value.KindOneofCase.ListValue)
+                {
+                    foreach (Value val in metadataValue.ListValue.Values)
+                    {
+                        valuesList.Add(val);
+                    }
+                }
+                else if (metadataValue.KindCase == Value.KindOneofCase.StringValue)
```

Comment:
Is it better to remove the string comparison and default to this `else` condition always?

@Ramanth

Reply:
Do you mean removing the conditional statements altogether and always doing `valuesList.Add(metadataValue)`? If so, I think that would be a step back because in cases where `metadataValue` is a list of values they would all be appended to `valuesList` as one.

However, if you mean changing the `else if` to a simple `else` and removing just that string comparison then I don't see an issue with that